File: src/SystemDomain.java
import Creator.CreateAdvisor;
import Creator.CreateCourse;
import Creator.CreateLecturer;
import Creator.CreatePage;
import Creator.CreateStudent;
import PersonObject.*;

import org.json.JSONException;
import java.io.IOException;
import java.util.ArrayList;

import Page.*;

public class SystemDomain {
    private CreateLecturer lecturerCreator;
    private CreateAdvisor advisorCreator;
    private CreateCourse courseCreator;
    private CreateStudent studentCreator;
    private CreatePage pageCreator;

    public SystemDomain() throws JSONException, IOException {
        lecturerCreator = new CreateLecturer("src\\JSON_Files\\lecturers.json");
        advisorCreator = new CreateAdvisor("src\\JSON_Files\\advisors.json",lecturerCreator.getLecturers());
        courseCreator = new CreateCourse("src\\JSON_Files\\courses.json",lecturerCreator.getLecturers());
        studentCreator = new CreateStudent("src\\JSON_Files\\Students\\","src\\JSON_Files\\student_json.txt",courseCreator.getCourses(),advisorCreator.getAdvisors());
        pageCreator = new CreatePage();
    }


    public CreateLecturer getLecturerCreator() {return lecturerCreator;}

    public void setLecturerCreator(CreateLecturer lecturerCreator) { this.lecturerCreator = lecturerCreator;}

    public CreateAdvisor getAdvisorCreator() {return advisorCreator;}

    public void setAdvisorCreator(CreateAdvisor advisorCreator) {this.advisorCreator = advisorCreator;}

    public CreateCourse getCourseCreator() {return courseCreator;}

    public void setCourseCreator(CreateCourse courseCreator) {this.courseCreator = courseCreator;}

    public CreateStudent getStudentCreator() {return studentCreator;}

    public void setStudentCreator(CreateStudent studentCreator) {this.studentCreator = studentCreator;}

	public CreatePage getPageCreator() {return pageCreator;}

	public void setPageCreator(CreatePage pageCreator) {this.pageCreator = pageCreator;}
}

File: iter3/python_development/System/SystemDomain.py
from iter3.python_development.Creator.CreateAdvisor import CreateAdvisor
from iter3.python_development.Creator.CreateCourse import CreateCourse
from iter3.python_development.Creator.CreateLecturer import CreateLecturer
from iter3.python_development.Creator.CreatePage import CreatePage
from iter3.python_development.Creator.CreateStudent import CreateStudent


class SystemDomain:
    def __init__(self):
        self.lecturerCreator = CreateLecturer("JSON_Files\\lecturers.json")
        self.advisorCreator = CreateAdvisor("JSON_Files\\advisors.json", self.lecturerCreator.getLecturers())
        self.courseCreator = CreateCourse("JSON_Files\\courses.json", self.lecturerCreator.getLecturers())
        self.studentCreator = CreateStudent("JSON_Files\\Students\\", "JSON_Files\\student_json.txt",
                                             self.courseCreator.getCourses(), self.advisorCreator.getAdvisors())
        self.pageCreator = CreatePage()

    def getLecturerCreator(self):
        return self.lecturerCreator

    def getAdvisorCreator(self):
        return self.advisorCreator

    def getCourseCreator(self):
        return self.courseCreator

    def getStudentCreator(self):
        return self.studentCreator

    def getPageCreator(self):
        return self.pageCreator

